apps/rehabstod/src/components/Form/SelectCareUnits.tsx

```diff
@@ -0,0 +1,48 @@
+import { IDSSelect } from '@frontend/ids-react-ts'
+import { Mottagning, UserPreferences, Vardenhet } from '../../schemas'
+import { useGetUserQuery } from '../../store/api'
+
+export function SelectCareUnits({ onChange, preferences }: { onChange: (value: string) => void; preferences: UserPreferences }) {
+  const { data: user } = useGetUserQuery()
+
+  function getUnits(): (Vardenhet | Mottagning)[] {
+    if (!user) {
+      return []
+    }
+    const units: (Vardenhet | Mottagning)[] = []
+    user.vardgivare.forEach((careProvider) => {
+      careProvider.vardenheter.forEach((careUnit) => {
+        units.push(careUnit)
+        if (careUnit.mottagningar && careUnit.mottagningar.length > 0) {
+          careUnit.mottagningar.forEach((reception) => units.push(reception))
+        }
+      })
+    })
+    return units
+  }
+
+  if (!user) {
+    return null
+  }
+
+  return (
+    <IDSSelect>
+      {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
```

Comment:
We can fix this rule to better work with IDS components. Please add this to `.eslintrc.cjs` rules.
```
'jsx-a11y/label-has-associated-control': [
  'error',
  {
    labelComponents: [],
    labelAttributes: [],
    controlComponents: [],
    assert: 'either',
    depth: 25,
  },
],
```

This copies the rules from airbnb but allows us to have label as sibling to the input https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/rules/react-a11y.js#L123-L129

https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/label-has-associated-control.md

The component also needs an ID between select and label, so please add `htmlFor={id}` to the label and `id={id}` to the select element. You can generate an ID with the React function `const id = useId()`

Reply:
Thanks for clarifying how to fix this issue. Worked great!